dbc_teacher.rb

```diff
@@ -0,0 +1,38 @@
+require_relative 'dbc_member'
+
+class DbcTeacher < DbcMember
+  attr_reader :salary, :target_raise
+  def initialize(options={})
+    super
+    @phase = 3
+  end
+
+  def set_phase(num)
+    @phase = num
+    "Cool, I've always wanted to teach phase #{num}!"
+  end
+
+  def salary=(new_salary)
+    puts "This better be good!"
+    @salary = new_salary
+  end
+
+  def receive_raise(raise)
+    @salary += raise
+  end
+
+  def set_performance_rating(rating)
+    response = ""
+    if rating > 90 && self.class == Teacher
+      receive_raise(@target_raise)
+      response = "Yay, I'm a great employee!"
+    elsif rating > 80 && self.class == ApprenticeTeacher
```

Comment:
Ahh, I see. I was seeing some people use constants to store the raise and the phase instead of instance variables. Would the refactor of this method look something like this? Also, is there a specific benefit between storing the rating as a constant versus instance variable?  

```ruby
  def set_performance_rating(rating)
    response = ""
    if rating > self.class::TARGET_RATING
    # Or if I used an instance variable
    # if rating > @target_rating
      receive_raise(@target_raise)
      response = "Yay, I'm a great employee!"
    else
      response += "Oh, well -- thanks to this actionable, specific, and kind "
      response += "feedback, I'll do better next time."
    end
    response
  end
end
```